src/main/java/carpet/settings/ParsedRule.java

```diff
@@ -105,7 +129,7 @@ else if (this.type.isEnum())
 
     public ParsedRule<T> set(ServerCommandSource source, String value)
     {
-        if (CarpetServer.settingsManager != null && CarpetServer.settingsManager.locked)
+        if (settingsManager != null && settingsManager.locked)
```

Comment:
adding reference to setting manager is a good change - this way each setting group can be managed independently